sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -775,6 +778,57 @@ def _get_sample_count(self, n_samples):
             return ceil(n_samples * self.subsample)
         return n_samples
 
+    def _get_color_dict_from_values(
+        self, value_list, categorical_or_continuous: str, cmap_name: str = "Spectral"
+    ):
+        color_dict = {}
+        if categorical_or_continuous == "categorical":
+            mapping = {
+                np.unique(value_list)[i]: f"C{i}"
+                for i in range(len(np.unique(value_list)))
+            }
+            color_dict["color_list"] = [
+                mapping[value_list[i]] for i in range(len(value_list))
+            ]
+            color_dict["categorical_mapping"] = mapping
+        elif categorical_or_continuous == "continuous":
+            cmap = mpl.cm.get_cmap(cmap_name)
+            norm = mpl.colors.Normalize(vmin=value_list.min(), vmax=value_list.max())
+            color_dict["color_list"] = [
+                mpl.colors.to_hex(cmap(norm(value))) for value in value_list
+            ]
+            color_dict["norm"] = norm
+            color_dict["cmap"] = cmap
+        return color_dict
+
+    def _check_if_colors_by_feature(self, ice_lines_kw, preds):
+        # case 1: color can be a single color for all lines given as a str for all lines
+        color_dict = {}
+        if isinstance(ice_lines_kw["color"], str):
+            color_dict["color_list"] = np.repeat(ice_lines_kw["color"], preds.shape[0])
+        # case 2: color can be given as a list values for which color is created
+        elif isinstance(ice_lines_kw["color"], (list, np.ndarray, pd.Series)):
+            value_list = np.array(ice_lines_kw["color"])
```

Comment:
bin immer noch nicht so ganz happy mit dem check, also weiß nicht, ob du da nicht noch was valides verpasst. vllt einfach in einem try/except block schauen ob du das als array umwandeln kannst und wenn ja ist gut? bzw sonst vllt noch gucken ob `iter(ice_lines_kw["color"])` probieren?